core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java

```diff
@@ -628,27 +628,27 @@ ReadResult run() {
                 objectsByClass.put(mapping.getKlass(), config.getConfigMapping(mapping.getKlass(), mapping.getPrefix()));
             }
 
-            Set<PropertyName> buildTimeNames = getMappingsNames(buildTimeMappings);
-            Set<PropertyName> buildTimeRunTimeNames = getMappingsNames(buildTimeRunTimeMappings);
-            Set<PropertyName> runTimeNames = getMappingsNames(runTimeMappings);
+            Map<PropertyName, String> buildTimeNames = mappingsToNames(buildTimeMappings);
+            Map<PropertyName, String> buildTimeRunTimeNames = mappingsToNames(buildTimeRunTimeMappings);
+            Map<PropertyName, String> runTimeNames = mappingsToNames(runTimeMappings);
```

Comment:
Could you clarify why this change? It doesn't look related to splitting the method?

Reply:
I dunno. It makes the intent less clear and I don't think you win anything by doing that, at least not a lot. I think what @franz1981 wanted to underline is that the keySet keeps the map around but given you switched to referencing the map directly, the only win is the very small keyset allocation. Given this is build time, I don't see the value of making the code more convoluted.